gammapy/modeling/models/spectral.py

```diff
@@ -560,8 +561,16 @@ def plot_error(
             ax.yaxis.set_units(DEFAULT_UNIT[sed_type] * energy.unit**energy_power)
 
         flux, flux_err = self._get_plot_flux(sed_type=sed_type, energy=energy)
-        y_lo = scale_plot_flux(flux - flux_err, energy_power).quantity[:, 0, 0]
-        y_hi = scale_plot_flux(flux + flux_err, energy_power).quantity[:, 0, 0]
+
+        if method == "log":
+            flux_err_log = flux_err.data / flux.data
+            y_lo = flux / np.exp(flux_err_log)
+            y_hi = flux * np.exp(flux_err_log)
+            y_lo = scale_plot_flux(y_lo, energy_power).quantity[:, 0, 0]
```

Comment:
You can even move the `y_lo` and `y_hi` out of the `if...else` block. Which should make it even more simpler...